main.rb

```diff
@@ -0,0 +1,173 @@
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+    user_choice = gets.chomp
+    start_options user_choice
+  end
+
+  def start_options(options)
+    case options
+    when '1'
+      list_books
+    when '2'
+      list_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rental_by_person_id
+    else
+      puts 'Exit'
+    end
+  end
+
+  def list_books
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    choices
+  end
+
+  def list_people
+    @people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, age: #{person.age}" }
+    choices
+  end
+
+  def create_person
+    print 'Do you want to create a student(1) or a teacher(2)? [Input the number]'
+    answer = gets.chomp
+
+    case answer
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Please choose a valid number'
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp.downcase == 'y'
+
+    student = Student.new(name: name, age: age, parent_permission: permission, classroom: @classroom)
+
+    @people << student
+
+    puts 'Person created successfully'
+    choices
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    teacher = Teacher.new(name: name, age: age, specialization: specialization)
+
+    @people << teacher
+
+    puts 'Person created successfully'
+    choices
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    book = Book.new(title, author)
+
+    @books << book
+
+    puts 'Book created successfully'
+
+    choices
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |book, idx| puts "#{idx}) Title: #{book.title}, Author: #{book.author}" }
+
+    book_index = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number(not id)'
+    @people.each_with_index { |person, idx| puts "#{idx}) ID: #{person.id}, Name: #{person.name}, Age: #{person.age}" }
+
+    person_index = gets.chomp.to_i
+
+    puts
+    print 'Date: '
+    date = gets.chomp
+
+    rental = Rental.new(date, @books[book_index], @people[person_index])
+
+    @rentals << rental
+
+    puts 'Rental created successfully'
+
+    choices
+  end
+
+  def list_rental_by_person_id
+    print 'ID of person: '
+    id = gets.chomp
+
+    puts 'Rentals: '
+
+    @rentals.each do |rental|
+      puts "Date: #{rental.date}, Book: '#{rental.book.title}' by #{rental.book.author}" if rental.person.id == id.to_i
+    end
+    choices
+  end
+end
```

Comment:
- [ ] According to the requirements, you are expected to create a separate `app.rb` file that will contain the App class you defined above. It should contain the methods I selected here.

  Basically, this `main.rb` file should only contain code related to listing the choices and running the program. Any other method for that builds the listing and creation of the books, person, or rental (like these ones) should be in the App file.